### ISSUE TITLE
Chat - Unable to remove emoji after inserting emoji using undo shortcut

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#31712]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.2-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause-Internal Team
**Slack conversation:** @

## Action Performed:
1. Navigate to staging.new.expensify.com.
2. Go to any chat.
3. Click on the emoji picker.
4. Select an emoji.
5. Press CMD+Z to undo the emoji selection.

## Expected Result:
The emoji will be removed.

## Actual Result:
The emoji cannot be removed using undo shortcut.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/cb3532af-4e3e-403d-97e3-ec59317f9bb0

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by s77rt** [comment]
@suneox Thanks for the proposal. I don't think your RCA is correct / complete. If we are updating the text without updating the history then I'd expect the undo action to skip a step and keep working. Also what about the case where the emoji is typed (and not selected from the picker)?

**Comment by s77rt** [comment]
@suneox Can you please elaborate your RCA first? How exactly is the history lost?

**Comment by shubham1206agra** [comment]
@suneox Please see https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand. `execCommand` is deprecated and is not recommended.

**Comment by s77rt** [comment]
@suneox Thanks for the clarification. Indeed this is a browser bug/limitation as I was able to reproduce it using just html [input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input). Can you please update your solution to cover both cases? (typing the emoji code and selecting it from the picker)

**Comment by s77rt** [comment]
@suneox Thanks for the update. I think the history approach is an overkill for a bug that is browser limitation related. `execCommand` is deprecated indeed but it may be a better option here. Can you please update your solution for the `execCommand` approach just in case we wanted to go that way.

**Comment by s77rt** [comment]
@suneox I think we should try explore more solutions. The history approach looks an overkill here and I'd prefer to avoid it.

**Comment by s77rt** [comment]
@suneox I would really prefer to avoid the history approach. Also please note that the solution should be explained in plain English without code diff, PRs, etc. (you can use pseudo code though)

**Comment by s77rt** [comment]
Not close Melvin. This is arguably a browser limitation/design. Still looking for proposals

**Comment by s77rt** [comment]
Asked on Slack https://expensify.slack.com/archives/C01GTK53T8Q/p1701901193272099

**Comment by s77rt** [comment]
Not overdue. Still discussing on slack

**Comment by mountiny** [comment]
Coming from the [Slack thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1701901193272099), this is a minor issue with a simple workaround and hard/ hacky fix so I am going to close this out
